themes/default/content/what-is/resolve-list-buckets-expired-token.md

```diff
@@ -0,0 +1,110 @@
+
+### Step 3: Create a new Pulumi ESC environment
+
+Once OIDC has been configured between Pulumi and AWS, the next steps is to create a new environment in the [Pulumi Cloud](https://app.pulumi.com/). Make sure that you have the correct organization selected in the left-hand navigation menu. From there, click the **Environments** link, then click the **Create environment** button. In the following pop-up, provide a name for your environment before clicking the **Create environment** button.
+
+{{< video title="Open environment in Pulumi ESC console" src="/what-is/esc-create-new-env.mp4" autoplay="true" loop="true" >}}
+
+### Step 4: Add the AWS provider integration
+
+Once you’ve created your new environment, you will be presented with a split-pane editor view. You’ll want to clear out the default placeholder content in the editor on the left-hand side and replace it with the following code, making sure to replace <your-oidc-iam-role-arn> with the value of your IAM role ARN from the configure OIDC step:
+
+```yaml
+values:
+  aws:
+    login:
+      fn::open::aws-login:
+        oidc:
+          duration: 1h
+          roleArn: <your-oidc-iam-role-arn>
+          sessionName: pulumi-environments-session
+  environmentVariables:
+    AWS_ACCESS_KEY_ID: ${aws.login.accessKeyId}
+    AWS_SECRET_ACCESS_KEY: ${aws.login.secretAccessKey}
+    AWS_SESSION_TOKEN: ${aws.login.sessionToken}
+```
+
+### Step 5: Run the aws CLI command
+
+With your environment set up, you can now validate your configuration. First make sure that your local environment does not have any AWS credentials configured. You can do this by running the `aws configure list` command as shown below:
+
+```bash
+$ aws configure list
+      Name                    Value             Type    Location
+      ----                    -----             ----    --------
+   profile                <not set>             None    None
+access_key                <not set>             None    None
+secret_key                <not set>             None    None
+    region                <not set>             None    None
+```
+
+Now run the `aws s3api list-buckets` command using `esc run` as shown below, making sure to replace `<your-pulumi-org-name>` and `<your-environment-name>` with the names of your own Pulumi organization and environment respectively:
+
+```bash
+esc run <your-pulumi-org-name>/<your-environment-name> -i aws s3api list-buckets --query "Buckets[].Name"
+```
+
+## Conclusion
+
+Pulumi ESC makes it easier than ever to tame infrastructure complexity, especially when running commands like `aws s3 ls`. Because Pulumi ESC supports dynamic credentials using OIDC across AWS, Azure, and Google Cloud, you no longer have to worry about credentials errors as the service will dynamically generate and refresh them for you. Check out the following links to learn more about Pulumi ESC today.
```

Comment:
there is an inconsistency here. you use s3 ls here but use s3api list-buckets elsewhere

Reply:
perhaps here we be even more specific. you no longer have to worry about ExpiredToken errors because ESC will dynamically generate and refresh your credential tokens for you...